core/shared/src/main/scala/fs2/internal/Algebra.scala

```diff
@@ -410,115 +410,72 @@ private[fs2] object Algebra {
 
     }
 
-  private def translateStep[F[_], G[_], X](
+  private def translateAux[F[_], G[_], O](
       fK: F ~> G,
-      next: FreeC[Algebra[F, X, ?], Unit],
+      stream: FreeC[Algebra[F, O, ?], Unit],
       concurrent: Option[Concurrent[G]]
-  ): FreeC[Algebra[G, X, ?], Unit] =
-    next.viewL match {
-      case _: FreeC.Result.Pure[Algebra[F, X, ?], Unit] =>
-        FreeC.pure[Algebra[G, X, ?], Unit](())
-
-      case failed: FreeC.Result.Fail[Algebra[F, X, ?], Unit] =>
-        Algebra.raiseError(failed.error)
+  ): FreeC[Algebra[G, O, ?], Unit] = {
 
-      case interrupted: FreeC.Result.Interrupted[Algebra[F, X, ?], _, Unit] =>
-        FreeC.interrupted(interrupted.context, interrupted.deferredError)
+    def translateStep[X](next: FreeC[Algebra[F, X, ?], Unit]): FreeC[Algebra[G, X, ?], Unit] =
+      next.viewL match {
+        case _: FreeC.Result.Pure[Algebra[F, X, ?], Unit] =>
+          FreeC.pure[Algebra[G, X, ?], Unit](())
 
-      case view: ViewL.View[Algebra[F, X, ?], y, Unit] =>
-        view.step match {
-          case output: Algebra.Output[F, X] =>
-            Algebra.output[G, X](output.values).transformWith {
-              case r @ Result.Pure(v) =>
```

Comment:
This is the one point in which there is a difference between `translate0` and `translateStep`.